### DISCUSSION TITLE
Using Top Level Await with TypeScript

### DISCUSSION THREAD

**Discussion by justsharan** [swc-project/swc#1441] [Q&A] [answered]
Title says it all... I'm trying to use top-level await with TypeScript, but I keep getting this error:
> unknown field `topLevelAwait`, expected one of `tsx`, `decorators`, `dynamicImport`, `importAssertions` at line 10 column 3

My `.swcrc` file:
```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false,
      "decorators": true,
      "dynamicImport": true,
      "topLevelAwait": true
    }
  }
}
```

**Comment by kdy1** [comment, marked as answer]
You need to set `jsc.target` to es2017 or upper.

I verified that the following `.swcrc` allows top-level awaits.

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript"
    },
    "target": "es2017"
  }
}

```

**Reply by justsharan** [reply]
Is there a way to suppress the error in VS Code? It's looking for the `target` option in the tsconfig file, but of course there isn't one since I'm using swc.
<img width="640" alt="image" src="https://user-images.githubusercontent.com/53885408/109425866-7ba0a100-79b8-11eb-8c89-995d3d20deff.png">

**Reply by kdy1** [reply]
I searched for it but failed to find the way.

Maybe support for `tsconfig.json` can be a solution, but it would require some time.
